lib/pbench/server/api/__init__.py

```diff
@@ -192,16 +191,15 @@ def get_server_config() -> PbenchServerConfig:
             f"{__name__}: ERROR: No config file specified; set" " _PBENCH_SERVER_CONFIG"
         )
 
-    try:
-        return PbenchServerConfig.create(cfg_name)
-    except BadConfig as e:
-        raise Exception(f"{__name__}: {e} (config file {cfg_name})").with_traceback(
-            e.__traceback__
-        )
+    return PbenchServerConfig.create(cfg_name)
+
 
+def create_app(server_config: PbenchServerConfig):
+    """Create Flask app with defined resource endpoints.
 
-def create_app(server_config):
-    """Create Flask app with defined resource endpoints."""
+    Returns:
+        A Flask application object on success
+    """
```

Comment:
Fixed.

Reply:
See corrections to the `Flask` app used in #3202.

Reply:
I'm not sure why I was supposed to look at #3202, but it made me look more closely at the code...and, I believe I've determined that, now that you've changed the function signature to match the docstring, the function signature was correct and it was the docstring which needed fixing.  🤷

Reply:
If you could point out what you think it should be rather than having me guess, that would be helpful, as I am not seeing where a change is needed.

Reply:
The docstring says that the function returns a "Flask application", but I didn't see a `return` statement with an object resembling a Flask app...did I miss it?

Also, it looks like the value returned by `create_app()` (to wit, `None`, I believe) is then returned by `app()`, which is extra "interesting"....

Reply:
Line 247 in the new code (228 in the old) is not a `return` statement?

Reply:
Having the shutdown hook in the middle is really inconvenient; we have some evidence that it's very easy and perhaps natural for the eye to stop at the "new method" declaration even though it's a local that's decorated into a cleanup role. Comments, docstring???

Reply:
Great suggestion for a PR!

Reply:
> Line 247 in the new code (228 in the old) is not a `return` statement?

My bad:  as Dave surmised, once I saw the function decorator at line 243, I mistakenly believed that I was at the end of the function body (and the diff's display eliding the rest of the body didn't disabuse me of the notion...).

Nevertheless, I've got to say that that use of a function decorator on the definition of an inner function is...well...too clever by half.  Aside from the fact that most people don't understand what function decorators really are or do, the effects in this usage aren't nearly as clear as the following (which I believe is equivalent):
```
    ...
    def shutdown_session(exception=None):
        Database.db_session.remove()
    ...
    app.teardown_appcontext(shutdown_session)
    ...
```
Among other things, this would allow the inner function to be defined near the top of the function body (instead of having to define it at the point where it is referenced, which was the basis for the confusion), and the call to `teardown_appcontext()`, positioned adjacent to the `try` block which calls `init_db()` would then make much more sense.

Alternatively, I believe this could be done in a single statement with a lambda:
```
    app.teardown_appcontext(lambda exception=None: Database.db_session.remove())
```
I'm not sure that that is better than the inner function, but it's definitely better than using the decorator.

In any case, the changes proposed in this PR are an improvement on the existing code.

Reply:
See PR #3212.